pkg/api/testing/fuzzer.go

```diff
@@ -429,6 +430,30 @@ func FuzzerFor(t *testing.T, version unversioned.GroupVersion, src rand.Source)
 				}
 			}
 		},
+		func(r *runtime.RawExtension, c fuzz.Continue) {
+			// Pick an arbitrary type and fuzz it
+			types := []runtime.Object{&api.Pod{}, &extensions.Deployment{}, &api.Service{}}
+			obj := types[c.Rand.Intn(len(types))]
+			c.Fuzz(obj)
+
+			// Find a codec for converting the object to raw bytes.  This is necessary for the
+			// api version and kind to be correctly set be serialization.
+			codec, err := testapi.GetCodecForObject(obj)
```

Comment:
This is broken for the round trip codecs - since we support encoding to protobuf, this would fail once we enforce the types match.  You need to be getting the current codec under test (which is determined by the test case itself).